### PR TITLE
[SPARK-38725][SQL][TESTS] Test the error class: DUPLICATE_KEY

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This pr aims to Add one test for the error class DUPLICATE_KEY to QueryParsingErrorsSuite, it's a followup of SPARK-37935.

### Why are the changes needed?
Add one test for the error class DUPLICATE_KEY to QueryParsingErrorsSuite

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

- Manual test：

```
./build/sbt "sql/testOnly *QueryParsingErrorsSuite*"
```

All tests passed.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala [hunk 1]
```diff
@@ -390,4 +390,19 @@ class QueryParsingErrorsSuite extends QueryTest with SharedSparkSession {
...
           |""".stripMargin)
   }
+
+  test("DUPLICATE_KEY: Found duplicate partition keys") {
+    validateParsingError(
+      sqlText = "INSERT OVERWRITE TABLE table PARTITION(p1='1', p1='1') SELECT 'col1', 'col2'",
+      errorClass = "DUPLICATE_KEY",
+      sqlState = "23000",
+      message =
+        s"""
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:400]:
```suggestion
        """
```

Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:400]: ok, i will fix it

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala [hunk 2]
```diff
@@ -390,4 +390,19 @@ class QueryParsingErrorsSuite extends QueryTest with SharedSparkSession {
...
+      errorClass = "DUPLICATE_KEY",
+      sqlState = "23000",
+      message =
+        """
+           |Found duplicate keys 'p1'(line 1, pos 29)
+           |
+           |== SQL ==
+           |INSERT OVERWRITE TABLE table PARTITION(p1='1', p1='1') SELECT 'col1', 'col2'
+           |-----------------------------^^^
+           |""".stripMargin)
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:406]: Could you fix indentation here, please. Should be 2 spaces.

Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:406]: ok, i will do it!

**Comment by MaxGekk** [comment]
@panbingkun BTW, it is better to follow the guide: https://spark.apache.org/contributing.html and create a separate branch for the PR, see:
Below steps will take your through the process.
3. Clone your fork and create a new branch
8. [Open a pull request](https://help.github.com/articles/using-pull-requests/) against the master branch of apache/spark

**Comment by panbingkun** [comment]
new pr: https://github.com/apache/spark/pull/36188, please review, Thanks

**Pull request closed without merging**